tests/testthat/test-tmb.R

```diff
@@ -1609,18 +1610,61 @@ test_that("h_mmrm_tmb works even when time point variable has unused factor leve
   tmp_data$AVISIT[1] <- "SCREENING" # nolint
   tmp_data$AVISIT <- as.factor(tmp_data$AVISIT) # nolint
 
-  result <- expect_silent(fit_mmrm(
+  expect_warning(result <- fit_mmrm(
     FEV1 ~ FEV1_BL + RACE + us(AVISIT | USUBJID),
     data = tmp_data,
     weights = rep(1, nrow(tmp_data))
-  ))
+  ), "In AVISIT there are dropped visits: SCREENING")
   expect_class(result, "mmrm_tmb")
   expect_identical(
     rownames(VarCorr(result)),
     c("VIS1", "VIS2", "VIS3", "VIS4")
   )
 })
 
+test_that("h_mmrm_tmb works if we keep the unused factor levels for specific covariance structure", {
+  # Create a data set where one visit level only has NA in the data.
```

Comment:
hm so looking at this now, we need to repeat this data manipulation. maybe nicer to already include an `AVISIT2` variable in the `fev_data` and do it there, document accordingly, and then just use here?

Reply:
Here I use a simpler approach of add a "SCREENING" level. I think now this is much simpler and no need to update the `fev_data` now.

Reply:
cool